### PR TITLE
MC-287 - Ivanti Service Manager - New Action - Create Service Request

### PR DESCRIPTION
**Pull request by dmandyna-r7**
## Proposed Changes

Ivanti Service Manager - New Action - Create Service Request

### Description

Describe the proposed changes:

  - New Create Service Request action

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

#### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Ivanti Service Manager:1.1.0. Step name: create_service_request\n",
    "meta": {},
    "output": {
      "service_request": {
        "@odata.context": "https://rapid7.vantosi.com/HEAT/api/odata/$metadata#servicereqs/$entity",
        "Cancelable": false,
        "CreatedBy": "Alan.Taylor",
        "CreatedDateTime": "2021-02-27T15:19:32-08:00",
        "CustomerLocation": "Chicago",
        "DefaultApprover": "RThomas@saasitdemo.com",
        "DefaultApprover_Valid": "AD770E6531B640879A49F3C40002C92F",
        "Email": "alan.taylor@saasitdemo.com",
        "EntityLink": "FA9C9DD75EF9455CBC892F65691A1E7F",
        "EntityLink_Category": "OrganizationalUnit",
        "EntityLink_RecID": "FA9C9DD75EF9455CBC892F65691A1E7F",
        "ForceNotCancelable": false,
        "ForceNotEditable": false,
        "HasScriptRun": false,
        "InShoppingCart": false,
        "IsDSMTaskExisted": false,
        "IsDraft": false,
        "IsEditable": true,
        "IsExternal": false,
        "IsInFinalState": false,
        "IsNewRecord": false,
        "IsNewRequestNotificationSent": false,
        "IsReportedByAlternateContact": false,
        "IsTemplateEditable": false,
        "IsUnRead": false,
        "IsVIP": false,
        "LastModBy": "Alan.Taylor",
        "LastModDateTime": "2021-02-27T15:19:32-08:00",
        "OrgUnitLink": "FA9C9DD75EF9455CBC892F65691A1E7F",
        "OrgUnitLink_Category": "OrganizationalUnit",
        "OrgUnitLink_RecID": "FA9C9DD75EF9455CBC892F65691A1E7F",
        "OrganizationUnitID": "Corporate IT Operations",
        "OwnerTeam": "Service Management Team",
        "OwnerTeamEmail": "servicemanagementteam@demokit.org",
        "OwnerTeam_Valid": "2E4BABD54FB9420D94F836F0D9B80C47",
        "Phone": "+1 (312) 9 3833",
        "Price": "0.0000",
        "Price_Currency": "USD",
        "ProfileFullName": "Alan Taylor",
        "ProfileLink": "1087342EA6954D7D96140D64B452E597",
        "ProfileLink_Category": "Employee",
        "ProfileLink_RecID": "1087342EA6954D7D96140D64B452E597",
        "ProfileLoginId": "Alan.Taylor",
        "ProgressBarPosition": "1",
        "ReadOnly": false,
        "RecId": "E554796248794BD8ABA1DC9F53CFC973",
        "SLA": "Benefits Management (Silver) for Corporate IT",
        "SLADisplayText": "Benefits Management (Silver) for Corporate IT",
        "SLALink": "A202D6F371F04629B2426A2BA32C4CB9",
        "SLALink_Category": "ServiceAgreement.SLA",
        "SLALink_RecID": "A202D6F371F04629B2426A2BA32C4CB9",
        "SendSurveyNotification": false,
        "Service": "Benefits Management",
        "ServiceReqNumber": 10209,
        "Service_Valid": "1A6C68571E6648E7A99305E1905C29B4",
        "Source": "Self Service",
        "Source_Valid": "B7065708DF4B449F8503621393B4A7AE",
        "Status": "Submitted",
        "Status_Valid": "E5BDDCB73AC74B7693A57AA1D84D42EB",
        "Subject": "Benefits Package Claim",
        "SvcReqTmplLink": "A299658A3EC84DC1BC401A065C0697D9",
        "SvcReqTmplLink_Category": "ServiceReqTemplate",
        "SvcReqTmplLink_RecID": "A299658A3EC84DC1BC401A065C0697D9",
        "Symptom": "This service request was created using InsightConnect",
        "TemplateCancelable": false,
        "TotalTimeSpent": 0,
        "Urgency": "Low",
        "Urgency_Valid": "320F22CAA2984C87B06AEFD3DE6FFBBF",
        "ivnt_PurchaseRequestCount": 0,
        "ivnt_RequestedCatalogItemCount": 0
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ivanti_service_manager:1.1.0 --debug run < tests/create_service_request1.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Ivanti Service Manager:1.1.0. Step name: create_service_request\n",
    "meta": {},
    "output": {
      "service_request": {
        "@odata.context": "https://rapid7.vantosi.com/HEAT/api/odata/$metadata#servicereqs/$entity",
        "Cancelable": false,
        "CreatedBy": "Alan.Taylor",
        "CreatedDateTime": "2021-02-27T15:19:41-08:00",
        "CustomerLocation": "Chicago",
        "DefaultApprover": "Julie.Curley@DemoKit.org",
        "DefaultApprover_Valid": "CD31ADF77BCA43A895A90952EE4AE520",
        "Email": "Josh.Mooney@DemoKit.org",
        "EntityLink": "374C95372668446EBB055D264BD69686",
        "EntityLink_Category": "OrganizationalUnit",
        "EntityLink_RecID": "374C95372668446EBB055D264BD69686",
        "ForceNotCancelable": false,
        "ForceNotEditable": false,
        "HasScriptRun": false,
        "InShoppingCart": false,
        "IsDSMTaskExisted": false,
        "IsDraft": false,
        "IsEditable": true,
        "IsExternal": false,
        "IsInFinalState": false,
        "IsNewRecord": false,
        "IsNewRequestNotificationSent": false,
        "IsReportedByAlternateContact": false,
        "IsTemplateEditable": false,
        "IsUnRead": false,
        "IsVIP": false,
        "LastModBy": "Alan.Taylor",
        "LastModDateTime": "2021-02-27T15:19:41-08:00",
        "OrgUnitLink": "374C95372668446EBB055D264BD69686",
        "OrgUnitLink_Category": "OrganizationalUnit",
        "OrgUnitLink_RecID": "374C95372668446EBB055D264BD69686",
        "OrganizationUnitID": "Nursing Services",
        "OwnerTeam": "Service Management Team",
        "OwnerTeamEmail": "servicemanagementteam@demokit.org",
        "OwnerTeam_Valid": "2E4BABD54FB9420D94F836F0D9B80C47",
        "Phone": "+1 (312) 9 3833",
        "Price": "0.0000",
        "Price_Currency": "USD",
        "ProfileFullName": "Josh Mooney",
        "ProfileLink": "DED86044C52845AF8B58AB120128BD84",
        "ProfileLink_Category": "Employee",
        "ProfileLink_RecID": "DED86044C52845AF8B58AB120128BD84",
        "ProfileLoginId": "Josh.Mooney",
        "ProgressBarPosition": "2",
        "ReadOnly": false,
        "RecId": "1D31B2E59140485187AA8C97CBB893B8",
        "SendSurveyNotification": true,
        "Service": "Desktop Service",
        "ServiceReqNumber": 10210,
        "Service_Valid": "164A2AB6ED554D80B3BE0B06C29DCC1A",
        "Source": "Self Service",
        "Source_Valid": "B7065708DF4B449F8503621393B4A7AE",
        "Status": "Approved",
        "Status_Valid": "782514675A8D40179C5088063344301D",
        "Subject": "Tablet Request",
        "SvcReqTmplLink": "429DCFE6EB05434B8EED6DE603CF36EF",
        "SvcReqTmplLink_Category": "ServiceReqTemplate",
        "SvcReqTmplLink_RecID": "429DCFE6EB05434B8EED6DE603CF36EF",
        "Symptom": "This service request was created using InsightConnect",
        "TemplateCancelable": false,
        "TotalTimeSpent": 0,
        "Urgency": "Medium",
        "Urgency_Valid": "44021B6CC6E44E598868C4B3306053AE",
        "ivnt_PurchaseRequestCount": 0,
        "ivnt_RequestedCatalogItemCount": 0
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ivanti_service_manager:1.1.0 --debug run < tests/create_service_request2.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Ivanti Service Manager:1.1.0. Step name: create_service_request\n",
    "meta": {},
    "output": {
      "service_request": {
        "@odata.context": "https://rapid7.vantosi.com/HEAT/api/odata/$metadata#servicereqs/$entity",
        "Cancelable": false,
        "CreatedBy": "Alan.Taylor",
        "CreatedDateTime": "2021-02-27T15:19:48-08:00",
        "CustomerLocation": "Frankfurt",
        "DefaultApprover": "Max.Lewis@DemoKit.org",
        "DefaultApprover_Valid": "E4D21DBCAC274E7F97A4A13CB1AC674B",
        "Email": "admin@saasitdemo.com",
        "EntityLink": "FA9C9DD75EF9455CBC892F65691A1E7F",
        "EntityLink_Category": "OrganizationalUnit",
        "EntityLink_RecID": "FA9C9DD75EF9455CBC892F65691A1E7F",
        "ForceNotCancelable": false,
        "ForceNotEditable": false,
        "HasScriptRun": false,
        "InShoppingCart": false,
        "IsDSMTaskExisted": false,
        "IsDraft": false,
        "IsEditable": true,
        "IsExternal": false,
        "IsInFinalState": false,
        "IsNewRecord": false,
        "IsNewRequestNotificationSent": false,
        "IsReportedByAlternateContact": false,
        "IsTemplateEditable": false,
        "IsUnRead": false,
        "IsVIP": false,
        "LastModBy": "Alan.Taylor",
        "LastModDateTime": "2021-02-27T15:19:48-08:00",
        "OrgUnitLink": "FA9C9DD75EF9455CBC892F65691A1E7F",
        "OrgUnitLink_Category": "OrganizationalUnit",
        "OrgUnitLink_RecID": "FA9C9DD75EF9455CBC892F65691A1E7F",
        "OrganizationUnitID": "Corporate IT Operations",
        "OwnerTeam": "Service Management Team",
        "OwnerTeamEmail": "servicemanagementteam@demokit.org",
        "OwnerTeam_Valid": "2E4BABD54FB9420D94F836F0D9B80C47",
        "Phone": "+49 69 8410 4935",
        "Price": "0.0000",
        "Price_Currency": "USD",
        "ProfileFullName": "Administrator Admin",
        "ProfileLink": "FB884D18F7B746A0992880F2DFFE749C",
        "ProfileLink_Category": "Employee",
        "ProfileLink_RecID": "FB884D18F7B746A0992880F2DFFE749C",
        "ProfileLoginId": "AAdmin",
        "ProgressBarPosition": "1",
        "ReadOnly": false,
        "RecId": "A40CA38FB813420C9AF63AA7E02A925B",
        "SendSurveyNotification": false,
        "Service": "Employee Administration",
        "ServiceReqNumber": 10211,
        "Service_Valid": "967FDE3BAB7F46AEAEF069D31E6BBE23",
        "Source": "Self Service",
        "Source_Valid": "B7065708DF4B449F8503621393B4A7AE",
        "Status": "Submitted",
        "Status_Valid": "E5BDDCB73AC74B7693A57AA1D84D42EB",
        "Subject": "Address Change",
        "SvcReqTmplLink": "D945CAB8F39A4D419F96FE59C6CE6FD4",
        "SvcReqTmplLink_Category": "ServiceReqTemplate",
        "SvcReqTmplLink_RecID": "D945CAB8F39A4D419F96FE59C6CE6FD4",
        "Symptom": "This service request was created using InsightConnect",
        "TemplateCancelable": false,
        "TotalTimeSpent": 0,
        "Urgency": "Medium",
        "Urgency_Valid": "44021B6CC6E44E598868C4B3306053AE",
        "ivnt_PurchaseRequestCount": 0,
        "ivnt_RequestedCatalogItemCount": 0
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ivanti_service_manager:1.1.0 --debug run < tests/create_service_request3.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nNo employees found. No employees found using data provided - Invalid User Name. Please validate and try again.",
    "log": "Connect: Connecting...\nrapid7/Ivanti Service Manager:1.1.0. Step name: create_service_request\nAn error occurred during plugin execution!\n\nNo employees found. No employees found using data provided - Invalid User Name. Please validate and try again.\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.2.0-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 326, in handle_step\n    output = self.start_step(\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.2.0-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 476, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.8/site-packages/ivanti_service_manager_rapid7_plugin-1.1.0-py3.8.egg/icon_ivanti_service_manager/actions/create_service_request/action.py\", line 22, in run\n    \"ProfileLink\": self.connection.ivanti_service_manager_api.search_employee(\n  File \"/usr/local/lib/python3.8/site-packages/ivanti_service_manager_rapid7_plugin-1.1.0-py3.8.egg/icon_ivanti_service_manager/util/api.py\", line 30, in search_employee\n    raise PluginException(\ninsightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!\n\nNo employees found. No employees found using data provided - Invalid User Name. Please validate and try again.\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ivanti_service_manager:1.1.0 --debug run < tests/create_service_request4_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nNo service request templates found. No service request templates found using data provided - Invalid Template Name. Please validate and try again.",
    "log": "Connect: Connecting...\nrapid7/Ivanti Service Manager:1.1.0. Step name: create_service_request\nAn error occurred during plugin execution!\n\nNo service request templates found. No service request templates found using data provided - Invalid Template Name. Please validate and try again.\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.2.0-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 326, in handle_step\n    output = self.start_step(\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.2.0-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 476, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.8/site-packages/ivanti_service_manager_rapid7_plugin-1.1.0-py3.8.egg/icon_ivanti_service_manager/actions/create_service_request/action.py\", line 20, in run\n    \"SvcReqTmplLink\": self.connection.ivanti_service_manager_api.search_service_request_template(\n  File \"/usr/local/lib/python3.8/site-packages/ivanti_service_manager_rapid7_plugin-1.1.0-py3.8.egg/icon_ivanti_service_manager/util/api.py\", line 60, in search_service_request_template\n    raise PluginException(\ninsightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!\n\nNo service request templates found. No service request templates found using data provided - Invalid Template Name. Please validate and try again.\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ivanti_service_manager:1.1.0 --debug run < tests/create_service_request5_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Ivanti Service Manager:1.1.0. Step name: create_service_request\n",
    "meta": {},
    "output": {
      "service_request": {
        "@odata.context": "https://rapid7.vantosi.com/HEAT/api/odata/$metadata#servicereqs/$entity",
        "Cancelable": false,
        "CreatedBy": "Alan.Taylor",
        "CreatedDateTime": "2021-02-27T15:20:03-08:00",
        "CustomerLocation": "Chicago",
        "DefaultApprover": "Julie.Curley@DemoKit.org",
        "DefaultApprover_Valid": "CD31ADF77BCA43A895A90952EE4AE520",
        "Email": "Josh.Mooney@DemoKit.org",
        "EntityLink": "374C95372668446EBB055D264BD69686",
        "EntityLink_Category": "OrganizationalUnit",
        "EntityLink_RecID": "374C95372668446EBB055D264BD69686",
        "ForceNotCancelable": false,
        "ForceNotEditable": false,
        "HasScriptRun": false,
        "InShoppingCart": false,
        "IsDSMTaskExisted": false,
        "IsDraft": false,
        "IsEditable": true,
        "IsExternal": false,
        "IsInFinalState": false,
        "IsNewRecord": false,
        "IsNewRequestNotificationSent": false,
        "IsReportedByAlternateContact": false,
        "IsTemplateEditable": false,
        "IsUnRead": false,
        "IsVIP": false,
        "LastModBy": "Alan.Taylor",
        "LastModDateTime": "2021-02-27T15:20:03-08:00",
        "OrgUnitLink": "374C95372668446EBB055D264BD69686",
        "OrgUnitLink_Category": "OrganizationalUnit",
        "OrgUnitLink_RecID": "374C95372668446EBB055D264BD69686",
        "OrganizationUnitID": "Nursing Services",
        "OwnerTeam": "Service Management Team",
        "OwnerTeamEmail": "servicemanagementteam@demokit.org",
        "OwnerTeam_Valid": "2E4BABD54FB9420D94F836F0D9B80C47",
        "Phone": "+1 (312) 9 3833",
        "Price": "0.0000",
        "Price_Currency": "USD",
        "ProfileFullName": "Josh Mooney",
        "ProfileLink": "DED86044C52845AF8B58AB120128BD84",
        "ProfileLink_Category": "Employee",
        "ProfileLink_RecID": "DED86044C52845AF8B58AB120128BD84",
        "ProfileLoginId": "Josh.Mooney",
        "ProgressBarPosition": "8",
        "ReadOnly": false,
        "RecId": "6B63A8C95FDC43B4BF6937FE9FDD6EE8",
        "SendSurveyNotification": false,
        "Service": "Desktop Service",
        "ServiceReqNumber": 10212,
        "Service_Valid": "164A2AB6ED554D80B3BE0B06C29DCC1A",
        "Source": "Self Service",
        "Source_Valid": "B7065708DF4B449F8503621393B4A7AE",
        "Status": "Waiting for Customer",
        "Status_Valid": "8A0FD3CB378A4E7F81F93CCCC144388B",
        "Subject": "Tablet Request",
        "SvcReqTmplLink": "429DCFE6EB05434B8EED6DE603CF36EF",
        "SvcReqTmplLink_Category": "ServiceReqTemplate",
        "SvcReqTmplLink_RecID": "429DCFE6EB05434B8EED6DE603CF36EF",
        "Symptom": "This service request was created using InsightConnect",
        "TemplateCancelable": false,
        "TotalTimeSpent": 0,
        "Urgency": "Medium",
        "Urgency_Valid": "44021B6CC6E44E598868C4B3306053AE",
        "ivnt_PurchaseRequestCount": 0,
        "ivnt_RequestedCatalogItemCount": 0
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ivanti_service_manager:1.1.0 --debug run < tests/create_service_request6.json
</summary>
</details>

#### Plugin Validation

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: plugin.spec.yaml[1451]: https://ivanti.example.com
violation: help.md[262]: https://example.com/HEAT/api/odata/$metadata#incidents/$entity
violation: help.md[430]: https://example.com/HEAT/api/odata/$metadata#incidents/$entity
[*] Plugin successfully validated!

----
[*] Total time elapsed: 12803.778ms
```

<summary>
icon-validate --all .
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 2122.16ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] For Markdown linting, install mdl: gem install mdl

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
Run started:2021-02-27 23:27:29.652971

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 3242
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check
```

<summary>
make validate
</summary>
</details>

### CODE REVIEW

File: ivanti_service_manager/plugin.spec.yaml [hunk 1]
```diff
@@ -734,6 +734,707 @@ types:
...
       type: boolean
       required: false
+  service_request:
+    RecId:
+      title: Record ID
+      description: Record ID
+      type: string
+      required: false
+    LastModDateTime:
+      title: Last Mod Date Time
```
Comment (jmcadams-r7) [inline ivanti_service_manager/plugin.spec.yaml:744]: Spell out Modified here. (Down below, they're using ESC, and that's spelled out in the title. Need to be consistent).

File: ivanti_service_manager/plugin.spec.yaml [hunk 2]
```diff
@@ -734,6 +734,707 @@ types:
...
+      description: Draft price
+      type: string
+      required: false
+    IsDraft:
+      title: Is Draft
+      description: Is draft
+      type: boolean
+      required: false
+    RequestOfferingWFTrigger:
+      title: Request Offering WF Trigger
```
Comment (jmcadams-r7) [inline ivanti_service_manager/plugin.spec.yaml:1229]: Spell out WF in the title.

**Review by jmcadams-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — DISMISSED** [review]

File: ivanti_service_manager/plugin.spec.yaml (hunk 1 above)
Comment (dmandyna-r7) [reply ivanti_service_manager/plugin.spec.yaml]: Thanks! That's updated now :)

File: ivanti_service_manager/plugin.spec.yaml (hunk 2 above)
Comment (dmandyna-r7) [reply ivanti_service_manager/plugin.spec.yaml]: I've updated the abbreviations in the titles, and ran `make regenerate` to update the schemas. Also pulled in latest master changes from master.

**Comment by dmandyna-r7** [comment]
@mrinehart-r7 / @jmcadams-r7 All the PR comments are now resolved 🙂 Thank you for the review!

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**